### PR TITLE
[SYCL][ESIMD] Enable the test for non variadic printf version

### PR DESCRIPTION
**Pull request by turinevgeny**

### CODE REVIEW

File: sycl/test-e2e/ESIMD/printf.cpp [hunk 1]
```diff
@@ -14,8 +14,8 @@
 // Test using __SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__ is disabled until ESIMD
 // supports it.
 // FIXME: enable that test
```
Comment (bader) [inline sycl/test-e2e/ESIMD/printf.cpp:16]: @turinevgeny, please, update the comments too.

File: sycl/test-e2e/ESIMD/printf.cpp [hunk 2]
```diff
@@ -14,8 +14,8 @@
...
-// RUNx: %{build} -D__SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__ -o %t_nonvar.out
-// RUNx: %{run} %t_nonvar.out | FileCheck %s
+// RUN: %{build} -D__SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__ -fsycl-device-code-split=per_kernel -o %t_nonvar.out
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/printf.cpp:17]: Just wondering, why do we need per_kernel splits?

**Comment by bader** [comment]
Maybe "[SYCL][ESIMD] Enable the test for non variadic printf version"?

File: sycl/test-e2e/ESIMD/printf.cpp (hunk 2 above)
Comment (turinevgeny) [reply sycl/test-e2e/ESIMD/printf.cpp:17]: Don't know, I aligned with the previous command line. It passes either way, I can remove it.

File: sycl/test-e2e/ESIMD/printf.cpp (hunk 1 above)
Comment (turinevgeny) [reply sycl/test-e2e/ESIMD/printf.cpp:16]: Yeah, thanks, missed it.

**Review by sarnex — APPROVED** [review]

**Comment by turinevgeny** [comment]
These failures are not related:

```
********************
Timed Out Tests (1):
  SYCL :: Basic/fill_accessor_pi.cpp

********************
Failed Tests (1):
  SYCL :: Basic/image/srgba-aspect-check.cpp

********************
Failed Tests (1):
  SYCL :: Graph/Explicit/queue_shortcuts.cpp
```

**Comment by turinevgeny** [comment]
@intel/llvm-gatekeepers, can you please merge this patch?

**Comment by steffenlarsen** [comment]
@turinevgeny - Could you please open an issue for the queue_shortcuts.cpp failure? I do not see any [issue](https://github.com/intel/llvm/issues) mentioning it. The HIP failures are indeed known (timeout can be spurious and the other is related to https://github.com/intel/llvm/issues/10460.)

**Comment by turinevgeny** [comment]
> @turinevgeny - Could you please open an issue for the queue_shortcuts.cpp failure? I do not see any [issue](https://github.com/intel/llvm/issues) mentioning it. The HIP failures are indeed known (timeout can be spurious and the other is related to #10460.)

After rerun today I do not see this problem (perhaps, it was resolved). Only failure is "Precommit", but I do not understand what exactly the problem is.

**Pull request merged**